aws/table_aws_spot_price_history.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+	"time"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSpotPriceHistory(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_spot_price_history",
+		Description: "AWS Spot Price History",
+		List: &plugin.ListConfig{
+			Hydrate: listSpotPriceHistory,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "availability_zone", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "product_description", Require: plugin.Optional},
+				{Name: "start_time", Require: plugin.Optional},
+				{Name: "end_time", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{Name: "availability_zone", Description: "The Availability Zone.", Type: proto.ColumnType_STRING},
+			{Name: "instance_type", Description: "The instance type.", Type: proto.ColumnType_STRING},
+			{Name: "product_description", Description: "A general description of the AMI.", Type: proto.ColumnType_STRING},
+			{Name: "spot_price", Description: "The maximum price per unit hour that you are willing to pay for a Spot Instance.", Type: proto.ColumnType_STRING},
+			{Name: "timestamp", Description: "The date and time the request was created", Type: proto.ColumnType_TIMESTAMP},
+			{Name: "start_time", Description: "The date and time, up to the past 90 days, from which to start retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("start_time")},
+			{Name: "end_time", Description: "The date and time, up to the current date, from which to stop retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("end_time")},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSpotPriceHistory(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_spot_price_history.listSpotPriceHistory", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxItems := int32(100)
```

Comment:
```suggestion
	maxItems := int32(1000)
```

According to https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSpotPriceHistory.html, it appears that the maximum value is 1000. When handling pagination, we set the page size to the maximum allowed (to reduce the number of API calls). For reference - https://steampipe.io/docs/develop/plugin-release-checklist#data-ingestion

Reply:
You're right ! I fix it